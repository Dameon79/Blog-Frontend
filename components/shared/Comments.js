import { Component } from 'react'

class Comments extends Component {
  render () {
    const { comment } = this.props
    return (
      <div className='c-comments-container__comment'>
        <img className='c-comments-container__comment-avatar' src='/static/images/logo.png' />
        <h3 className='c-comments-container__comment-name'>{comment.name}</h3>
        <h3 className='c-comments-container__comment-body'>{comment.body}</h3>
        <h3 className='c-comments-container__comment-created'>{comment.createdAt}</h3>
      </div>
    )
  }
}

export default Comments
