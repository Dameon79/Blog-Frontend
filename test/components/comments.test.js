import Comments from '../../components/shared/Comments'

describe('<Comments />', () => {
  it('it renders correctly', () => {
    const component = shallow(
      <Comments />
    )

    expect(component).toMatchSnapshot()
  })
})