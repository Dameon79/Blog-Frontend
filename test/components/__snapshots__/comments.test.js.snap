// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> it renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comments />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="c-comments-container__comment-counter"
        >
          1 Comment
        </h3>,
        <a
          className="c-comments-container__login"
          href="/auth/auth0"
        >
          Login to comment
        </a>,
        <div
          className="c-comments-container__comment"
        >
          <img
            className="c-comments-container__comment-avatar"
            src="/static/images/logo.png"
          />
          <h3
            className="c-comments-container__comment-name"
          >
            Dameon Sumpton
          </h3>
          <h3
            className="c-comments-container__comment-body"
          >
            Sample Comment
          </h3>
          <h3
            className="c-comments-container__comment-created"
          >
            Posted Today
          </h3>
        </div>,
        <script
          src="node_modules/auth0-js/build/auth0.js"
          type="text/javascript"
        />,
      ],
      "className": "c-comments-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "1 Comment",
          "className": "c-comments-container__comment-counter",
        },
        "ref": null,
        "rendered": "1 Comment",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login to comment",
          "className": "c-comments-container__login",
          "href": "/auth/auth0",
        },
        "ref": null,
        "rendered": "Login to comment",
        "type": "a",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="c-comments-container__comment-avatar"
              src="/static/images/logo.png"
            />,
            <h3
              className="c-comments-container__comment-name"
            >
              Dameon Sumpton
            </h3>,
            <h3
              className="c-comments-container__comment-body"
            >
              Sample Comment
            </h3>,
            <h3
              className="c-comments-container__comment-created"
            >
              Posted Today
            </h3>,
          ],
          "className": "c-comments-container__comment",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "c-comments-container__comment-avatar",
              "src": "/static/images/logo.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dameon Sumpton",
              "className": "c-comments-container__comment-name",
            },
            "ref": null,
            "rendered": "Dameon Sumpton",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sample Comment",
              "className": "c-comments-container__comment-body",
            },
            "ref": null,
            "rendered": "Sample Comment",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Posted Today",
              "className": "c-comments-container__comment-created",
            },
            "ref": null,
            "rendered": "Posted Today",
            "type": "h3",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "src": "node_modules/auth0-js/build/auth0.js",
          "type": "text/javascript",
        },
        "ref": null,
        "rendered": null,
        "type": "script",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="c-comments-container__comment-counter"
          >
            1 Comment
          </h3>,
          <a
            className="c-comments-container__login"
            href="/auth/auth0"
          >
            Login to comment
          </a>,
          <div
            className="c-comments-container__comment"
          >
            <img
              className="c-comments-container__comment-avatar"
              src="/static/images/logo.png"
            />
            <h3
              className="c-comments-container__comment-name"
            >
              Dameon Sumpton
            </h3>
            <h3
              className="c-comments-container__comment-body"
            >
              Sample Comment
            </h3>
            <h3
              className="c-comments-container__comment-created"
            >
              Posted Today
            </h3>
          </div>,
          <script
            src="node_modules/auth0-js/build/auth0.js"
            type="text/javascript"
          />,
        ],
        "className": "c-comments-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "1 Comment",
            "className": "c-comments-container__comment-counter",
          },
          "ref": null,
          "rendered": "1 Comment",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login to comment",
            "className": "c-comments-container__login",
            "href": "/auth/auth0",
          },
          "ref": null,
          "rendered": "Login to comment",
          "type": "a",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="c-comments-container__comment-avatar"
                src="/static/images/logo.png"
              />,
              <h3
                className="c-comments-container__comment-name"
              >
                Dameon Sumpton
              </h3>,
              <h3
                className="c-comments-container__comment-body"
              >
                Sample Comment
              </h3>,
              <h3
                className="c-comments-container__comment-created"
              >
                Posted Today
              </h3>,
            ],
            "className": "c-comments-container__comment",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "c-comments-container__comment-avatar",
                "src": "/static/images/logo.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dameon Sumpton",
                "className": "c-comments-container__comment-name",
              },
              "ref": null,
              "rendered": "Dameon Sumpton",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sample Comment",
                "className": "c-comments-container__comment-body",
              },
              "ref": null,
              "rendered": "Sample Comment",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Posted Today",
                "className": "c-comments-container__comment-created",
              },
              "ref": null,
              "rendered": "Posted Today",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "node_modules/auth0-js/build/auth0.js",
            "type": "text/javascript",
          },
          "ref": null,
          "rendered": null,
          "type": "script",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
